src/test/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquareTest.java

```diff
@@ -0,0 +1,119 @@
+        notSquare = new MagicSquare(new int[][]{{4, 14, 12, 1}, {9, 7, 8, 12, 9}, {5, 0, 10, 99}, {16, 2, 9, 13}, {8}});
+        negativeValueNotSquare = new MagicSquare(new int[][]{{4, 14, 12, 1}, {9, 7, 8, 12, 9}, {5, 0, 10, -7}, {16, 2, 9, 13}, {8}});
+        emptyValue = new MagicSquare(new int[][]{});
+        oneValue = new MagicSquare(new int[][]{{1}});
+    }
+
+    @Test
+    void checkIsMagicSquare() {
+        assertThat("Check that biDimensional array is Magic Square", magicSquare.isMagicSquare(),
+                is(true));
+    }
+
+    @Test
+    void checkIsNotMagicSquare() {
+        List<MagicSquare> squareList = new ArrayList<>();
+        squareList.addAll(Arrays.asList(notSquare, emptyValue, oneValue, notMagicSquare, negativeValueNotSquare));
+        for (MagicSquare square : squareList) {
+            assertThat("Check that biDimensional array is NOT Magic Square", square.isMagicSquare(),
+                    is(false));
+        }
+    }
+
+
+    @Test
+    void checkIsAllValuesPositiveIntoSquare() {
+        List<MagicSquare> squareList = new ArrayList<>();
+        squareList.addAll(Arrays.asList(notMagicSquare, magicSquare));
+        for (MagicSquare square : squareList) {
+            assertThat("Check that biDimensional array all values are positive", square.isPositiveAllValuesIntoSquare(),
+                    is(true));
+        }
+    }
+
+    @Test
+    void checkIsAtLeastOneNegativeValueIntoMagicSquare() {
+        assertThat("Check that at least one value is negative or 0", negativeValueNotSquare.isPositiveAllValuesIntoSquare(),
+                is(false));
+    }
+
+    @Test
+    void checkIsSquare() {
+        List<MagicSquare> squareList = new ArrayList<>();
+        squareList.addAll(Arrays.asList(magicSquare, notMagicSquare));
+        for (MagicSquare square : squareList) {
+            assertThat("Check that biDimensional array is Square", square.isSquare(),
+                    is(true));
+        }
+    }
+
+    @Test
+    void checkIsNotSquare() {
+        List<MagicSquare> squareList = new ArrayList<>();
+        squareList.addAll(Arrays.asList(emptyValue, oneValue, notSquare, negativeValueNotSquare));
+        for (MagicSquare square : squareList) {
+            assertThat("Check that biDimensional array is not Square", notSquare.isSquare(),
+                    is(false));
+        }
+    }
+
+    @Test
+    void checkIsNotEmpty() {
+        List<MagicSquare> squareList = new ArrayList<>();
+        squareList.addAll(Arrays.asList(magicSquare, notMagicSquare, oneValue, notSquare, negativeValueNotSquare));
+        for (MagicSquare square : squareList) {
+            assertThat("Check that biDimensional array is not Empty", square.isNotEmpty(),
+                    is(true));
+        }
+    }
+
+    @Test
+    void checkIsEmptySquare() {
+        assertThat("Check that biDimensional array is Empty", emptyValue.isNotEmpty(),
+                is(false));
```

Comment:
This assert is a real roller coaster... Could you please simplify it to something like: 
`assertTrue("Check that biDimensional array is Empty", emptyValue.isEmpty());`

Reply:
I renamed isNotEmpty into isNotEmptyArray, to be more clear what method is tested